src/Components/Aspire.Microsoft.Data.SqlClient/AspireSqlServerSqlClientExtensions.cs

```diff
@@ -19,6 +19,134 @@ public static class AspireSqlServerSqlClientExtensions
 {
     private const string DefaultConfigSectionName = "Aspire:Microsoft:Data:SqlClient";
 
+    private static string EscapeIdentifier(string identifier) => identifier.Replace("]", "]]");
+
+    /// <summary>
+    /// Opens a database connection with the property settings specified by the <see cref="SqlConnection.ConnectionString"/>. If the database does not exist it attempts to create it.
+    /// </summary>
+    /// <param name="connection">The <see cref="Microsoft.Data.SqlClient.SqlConnection" />.</param>
+    /// <remarks>
+    /// <para>
+    /// This method wraps <see cref="Microsoft.Data.SqlClient.SqlConnection.Open()"/> but adds exception handling to detect
+    /// a <see cref="SqlException"/> with a <see cref="SqlException.Number"/> of 4060 (Database not found). If this exception is
+    /// throw the connection is closed and the connection string updated to connect to the <c>master</c> database.
+    /// </para>
+    /// <para>
+    /// If database creation succeeds then the connection is changed to use the original database, if creation fails then
+    /// the original exception is thrown.
+    /// </para>
+    /// </remarks>
+    public static void OpenWithCreate(this SqlConnection connection)
```

Comment:
I mean this isn't public API we would expose right? This should be an ADO.NET thing.

Reply:
I said the same in https://github.com/dotnet/aspire/issues/4160#issuecomment-2108101239.

(not sure where the conversation should happen - the issue or the PR)

Reply:
Do it in the issue. This was just so we had something concrete to talk about.

And I agree this should be something in ADO.NET